File: ShapeRain/src/shape/Circle.java
package shape;

import processing.core.PApplet;
import processing.core.PVector;

public class Circle extends Shape {
    private float radius;

    public Circle(PApplet p, PVector position, float radius) {
        super(p, position);
        this.radius = radius;
    }

    public void display() {
        p.noStroke();
        p.fill(color.getColor());
        p.pushMatrix();
        p.translate(position.x, position.y);
        p.ellipse(0, 0, radius * 2, radius * 2);
        p.popMatrix();
    }

}

File: docs/ShapeRain/shape/circle.js
class Circle extends Shape {

    constructor(position, color, radius) {
        super(position, color);
        this.radius = radius;
    }

    display() {
        push();
        noStroke();
        fill(this.color.getColor());
        translate(this.position.x, this.position.y);
        ellipse(0, 0, this.radius * 2, this.radius * 2);
        pop();
    }
}
